src/MergeContext.js

```diff
@@ -108,6 +108,82 @@ class Approval {
     }
 }
 
+class StatusCheck
+{
+    constructor(raw) {
+        assert(raw.context);
+        assert(raw.state);
+        assert(raw.target_url);
+        assert(raw.description);
+
+        this.context = raw.context;
+        this.state = raw.state;
+        this.targetUrl = raw.target_url;
+        this.description = raw.description;
+
+        this._raw = raw;
+    }
+}
+
+// Passed required status checks analysis
+class StatusResult
+{
+    // checksNumber:
+    //   for staged commits: the bot-configured number of required checks (Config.stagingChecks()),
+    //   for PR commits: GitHub configured number of required checks (requested from GitHub)
+    // context: either "PR" or "Staging";
+    constructor(checksNumber, context) {
+        assert(checksNumber !== undefined);
+        assert(checksNumber !== null);
+        assert(context);
+        this.checksNumber = checksNumber;
+        this.context = context;
+        this.requiredStatuses = [];
+    }
+
+    addRequiredStatus(requiredStatus) {
+        assert(requiredStatus);
+        this.requiredStatuses.push(requiredStatus);
+    }
+
+    // no more required status changes or additions are expected
+    final() {
+        return (this.requiredStatuses.length >= this.checksNumber) &&
+            this.requiredStatuses.every(check => check.state !== 'pending');
+    }
+
+    // something went wrong with at least one of the required status checks
+    failed() {
+        return this.requiredStatuses.some(check => check.state !== 'pending' && check.state !== 'success');
+    }
+
+    // the results are final and all checks were a success
+    succeeded() {
+        return this.final() && this.requiredStatuses.every(check => check.state === 'success');
+    }
+
+    toString() {
+        let combinedStatus = "context: " + this.context + " expected/received: " + this.checksNumber + "/" +
+            this.requiredStatuses.length + ", combined: ";
+        if (this.failed())
+            combinedStatus += "failure";
+        else if (this.succeeded())
+            combinedStatus += "success";
+        else if (!this.final())
+            combinedStatus += "to-be-determined";
+        else
+            combinedStatus += "unexpected";
```

Comment:
I would reorder the last two and add a trailing comment to clarify the final() case: `// final() implies failed() or succeeded()`